### ISSUE TITLE
[Telemetry] Sync history and bookmarks

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#6396]
#### Description & Product Manager / Data Scientist User Story
- As a product owner, I want to know if users choose to enable sync for some data and not others, so I can plan future sync related features accordingly. 

#### Hypothesis
- We believe that users want the ability to control what type of personal data is synced across devices. 

#### What questions will you answer with this data?
- How many users turn off sync for either bookmarks or history?
- How many users do not interact with Sync settings?

#### Why does Mozilla need to answer these questions?
- To understand user behavior and plan future enhancements accordingly. 

#### What probes (suggested, if applicable)
- Sync history
- Sync bookmarks

### Acceptance Criteria (Added by PM)
- Event pings can be queried via re:dash
- We are sending telemetry events for the actions listed in the requirements
- We have documented the telemetry
- We have asked a data steward to [review](https://github.com/mozilla/data-review/blob/master/request.md) the telemetry
- NOT an AC: Data science to create dashboard or further graphs (this will be a separate issue, this issue is only about hooking up the events described and that they can be queried in re-dash)

**Comment by AndiAJ** [comment]
Hi, please refer to this [comment](https://github.com/mozilla-mobile/fenix/issues/3060#issuecomment-522444633) as it was already working since Aug.

I've re-tested on the latest **Nightly Build Nightly Build #13240608 from 20/11** using a Google Pixel 3a XL (Android 9)

✔️ Auto Sync right after log in:
Bookmarks - Ping 40b5b6ce-3be2-42ca-971f-819218a4d30e
History - Ping 064512ae-0474-4848-b067-e969fcdc787a

✔️ Manually performing a sync having both options enabled
Bookmarks - Ping 23cb51c5-e75e-46dc-8f2d-2dd933059c5a
History - Ping c20b0bcf-4593-4e98-87d6-70197955df24

✔️ Individual sync
Bookmarks - Ping 9ab8a139-6c8b-4a44-a660-d01dbc49a322
History - Ping c5d8ce27-d705-45b1-893f-0e19392268d5

[Logcat](https://github.com/mozilla-mobile/fenix/files/3870097/bookhist.txt)
[Glean dashvoard](https://debug-ping-preview.firebaseapp.com/pings/aj-t2011000)

@sblatz - The [matter](https://github.com/mozilla-mobile/fenix/issues/3060#issuecomment-522627653) regarding the sync events getting triggered twice after sign in caused by an artifact isn't there anymore. :relaxed:
@vesta0  - Please share your thought on this

**Comment by AndiAJ** [comment]
Hi, I've re-checked this matter on the latest **Nightly Build #13440604 from 12/10** using a Google Pixel 3a (Android 9)

✔️ Individual sync
Bookmarks - Ping 202059db-3baa-4074-9b56-c22d02919d21
History - Ping b771792a-617c-4fee-8c86-febe6a0ff4de

✔️ Manually performing a sync having both options enabled
Bookmarks - Ping 527e2dad-581e-4025-a0ea-f5e6795be7e4
History - Ping 653de544-f8ad-4d42-b5ca-7ebfe00a141f

❓ Bookmarks pref
Pref Bookmarks OFF a0dd175e-d54c-4819-a094-fe9e08adeedd
Pref Bookmarks ON 1064e5b0-f0c0-486c-8742-5a2b1802b9c6

❓History pref
Pref History OFF af0f4933-6b33-495e-9ca4-c9e95575d364
Pref History ON 6754682d-0419-4cec-bea4-ad0d34679b23

[Logcat](https://github.com/mozilla-mobile/fenix/files/3945983/BH.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/aj-test1012002)

There's some inconsistency regarding the **Bookmarks and History pref ON/OFF event type pings**. It seems that the corresponding event type ping is generated 2 times.


► Event type ping after unchecking History 
```
"events": [
        {
          "category": "events",
          "extra": {
            "enabled": "false",
            "preference_key": "pref_key_sync_history"
          },
          "name": "preference_toggled",
          "timestamp": 0
        },
        {
          "category": "events",
          "extra": {
            "enabled": "false",
            "preference_key": "pref_key_sync_history"
          },
          "name": "preference_toggled",
          "timestamp": 1
        }

```

► Event type ping after unchecking Logins (#5586 )
```
"events": [
        {
          "category": "events",
          "extra": {
            "enabled": "false",
            "preference_key": "pref_key_sync_logins"
          },
          "name": "preference_toggled",
          "timestamp": 0
        }
```

**I'll remove the QA needed label until further notice.**

**Comment by liuche** [comment]
We'll keep this one in the sprint.

**Comment by ekager** [comment]
@AndiAJ could we verify that we are no longer sending the pings 2 times?

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on **Fenix Beta 3.2.0-beta. 2 #20230233 from 01/23** using a Google Pixel 3a XL (Android 10)

✔️ Auto Sync right after log in:
Bookmarks - Ping 63cecb95-fb8e-43c3-b655-66f9d675888f
History - Ping 867c1ef0-14d4-42a8-b7ef-5c86398cc60c
Logins - Ping 53aee8c9-2139-45c7-8082-20f5e003c868


✔️ Manually performing a sync having both options enabled
Bookmarks - Ping bc506f3e-7d52-4510-8600-90c287570ee6
History - Ping fd510694-58cd-430f-a808-39c8efce8a14
Logins - Ping a1a35413-4c8f-47e1-b02c-fb6349f62e5c

✔️ Individual sync
Bookmarks - Ping 75dc6950-fba9-411f-83c8-14d7b4a1c91d
History - Ping - Ping c597fef4-bb11-4387-b30e-f1e4238cf109
Logins - Ping 5632d0f7-b527-444a-9b8a-f9015c3302cc (disabled the 2 other ones)

✔️ Bookmarks pref
Pref Bookmarks OFF - Ping 1c72201c-327b-4ce4-837d-709c915f89cc
Pref Bookmarks ON - Ping 77bf847e-15d7-45d6-a01b-d7711a19fa7c

✔️ History pref
Pref History OFF - Ping 2eef3749-3383-46aa-8710-0152df7ce82e
Pref History ON - Ping 530b3af2-1d6f-45bb-8f31-22e7b9b08221

[Logcat](https://github.com/mozilla-mobile/fenix/files/4102431/B.HSync.txt)
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/aj-t2301100000)

👉 **The Logins Sync works properly on the Beta** build, on the other hand, due to the fact that it's **behind a Nightly flag** I wasn't able to test the **Pref switch**.
Because of this, I've done it on the **latest Nightly Build #20230608 from 1/23**

✔️ Logins pref
Pref Logins OFF - Ping e5ee6278-5f9f-4e71-86d8-6b873d786732
Pref Logins ON - Ping 1a3dd7d2-8d58-427b-8818-50195536ab84

[Logcat](https://github.com/mozilla-mobile/fenix/files/4102440/LOnOff.txt)
[Glean dahboard](https://debug-ping-preview.firebaseapp.com/pings/aj-t230101010)
